### DISCUSSION TITLE
Dependency on Tensorflow

### DISCUSSION THREAD

**Discussion by nanofi** [google/flax#420] [General] [answered]
Hi,

Currently, only the code `flax/flax/metrics/tensorbord.py` depends on TensorFlow. We may use flax without installing TensorFlow if we isolate the TensorBoard feature to an extra library. Do you have a plan for removing this dependency on TensorFlow? TensorFlow is a huge library and has a high loading cost. So, I think it is nice to remove the dependency on TensorFlow.

**Comment by jheek** [comment, marked as answer]
You can use Flax without TensorFlow but not tensorboard logging. Ideally we would also like to support tensorboard logging with a small independent library for writing the metrics in the appropriate format. This would have to be provided by the TensorFlow/Tensorboard team though.

**Comment by nanofi** [comment]
Thank you for your prompt response! I've confirmed Flax works without TensorFlow except tensorboard feature. I asked this question because I wondered why Flax has this feature even though it is not crucial but introduces the dependency on the heavy library TensorFlow.
